tensorflow_addons/image/color_ops.py

```diff
@@ -132,3 +133,172 @@ def sharpness(
         image = to_4D_image(image)
         image = _sharpness_image(image, factor=factor)
         return from_4D_image(image, image_dims)
+
+
+def _clahe(
+    image: TensorLike, clip_limit: Number, tile_grid_size: Union[List[int], Tuple[int]]
+) -> tf.Tensor:
+    """Implements CLAHE as tf ops"""
+    original_2d_shape = (tf.shape(image)[0], tf.shape(image)[1])
+    original_dtype = image.dtype
+
+    # Need image in int32 format for later gather_nd ops
+    image = tf.cast(image, tf.int32)
```

Comment:
Are you sure that having a so early casting to `tf.int32` is ok?

Reply:
The first thing that requires a cast to int32 is `tf.math.bincount` (similar to [tfa.image.equalize](https://github.com/tensorflow/addons/blob/77aa837bba0fb88d5a5cb1018c9fcc162726370e/tensorflow_addons/image/color_ops.py#L34)). It could be pushed later, but doing so doesn't seem to affect performance.

If you would like the cast pushed later for other reasons, though, I can definitely push that change.